### PR TITLE
Add support for Python 3.5

### PR DESCRIPTION
**Pull request by AndreLouisCaron**
This is a redux of #1178.  It completes the port to make the build pass.

There are a few subtle changes in here, so I split each compatibility issue into its own commit.  It might be easier to review them one by one.

@jarondl Feel free to merge this commits into your PR.  I don't mind which one gets merged in the end :-)

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -1,11 +1,9 @@
 language: python
 python:
 - '2.7'
-env:
-- TOXENV=docs
-- TOXENV=py27
+- '3.5'
```
Comment (AndreLouisCaron) [inline .travis.yml]: **TODO**: make sure we can build the docs again.

Comment (jarondl) [inline .travis.yml]: What about `3.6`?

File: tox.ini [hunk 2]
```diff
@@ -1,6 +1,6 @@
 [tox]
 project = elastalert
-envlist = py27,docs
+envlist = py27,py35,docs
```
Comment (jarondl) [inline tox.ini]: again, what about `py36`?

File: .travis.yml (hunk 1 above)
Comment (AndreLouisCaron) [inline .travis.yml]: I wanted to try to minimize the scope of the changes.  Once this is merged, adding 3.6 should be a breeze.

Comment (AndreLouisCaron) [inline .travis.yml]: Added Python 3.6 to the list.

**Comment by jarondl** [comment]
@AndreLouisCaron What about PyPI classifiers? see   AndreLouisCaron/elastalert#1  and
https://github.com/AndreLouisCaron/elastalert/commit/b6fef27e579e2df1c8d257ce60a9883adc2fd021.patch


@Qmando   This PR adds Python3.5 and Python3.6 without harming Python2.7 (based on the test-suite). Any chances to merge this?

**Comment by AndreLouisCaron** [comment]
@jarondl Just added the classifiers, thanks for reminding me :-)

File: .travis.yml (hunk 1 above)
Comment (gzeronet) [reply .travis.yml]: :+1:

**Comment by CodingSpiderFox** [comment]
Any progress on this one?

**Comment by darrenhaken** [comment]
How's this coming along?

**Comment by sergekir** [comment]
Need some help with this?

**Comment by AndreLouisCaron** [comment]
I'm not sure I need any help.  I thought I had something that would pass, but it was never merged by maintainers...  If we can get their support with intent of merging, I guess I could do a pass to fix conflicts. -- should be rather easy.

**Comment by wlabatey** [comment]
+1 Any news on this? Would be great to be able to use Python 3.

**Comment by q2dg** [comment]
PLEASE!!

**Comment by Salaander** [comment]
Would be really nice to have python3 support. Tell me if I can do something to support this.

**Comment by aboese** [comment]
Some basic guess work (so willing to be wrong), but maybe go with an auth library that is already 3+ compatible?  google.auth comes to mind.

**Comment by aboese** [comment]
Also, create_index.py only needs Auth for configs that really use auth. Almost anyone trying to use create_index.py with python 3 without auth could probably just change all the raw_input to input, comment out all the Auth related lines as well as the last 5 or so lines on the ElasticSearch call. Easy peasy.

**Comment by aboese** [comment]
Maybe you'll have to temporarily drop Stride support?

**Comment by clearyusc** [comment]
any update on this?

**Comment by nleo** [comment]
Hi, guys!

It already a middle of 2019 here and I have no python2 on my servers now. Also I have no reason to install it.

**Comment by Qmando** [comment]
@nleo check out branch `beta_py3` if you need it now

**Comment by GoodMirek** [comment]
I feel like this PR is obsolete due to already merged #2285.

**Comment by AndreLouisCaron** [comment]
Closing in favor of #2285.  Thanks to @Qmando for pushing this through :-)

**Pull request closed without merging**